modules/apps/portal/portal-dao-orm-custom-sql-impl/src/main/java/com/liferay/portal/dao/orm/custom/sql/internal/CustomSQLImpl.java

```diff
@@ -213,6 +216,37 @@ public String get(
 		return sql;
 	}
 
+	@Override
+	public Predicate getKeywordsPredicate(
+		Expression<String> expression,
+		BiFunction<Expression<String>, String, Predicate> operatorBiFunction,
+		String[] values) {
+
+		if ((values != null) && (values.length == 0)) {
```

Comment:
@Preston-Crary do you mean ```if ((values == null) || (values.length == 0)) {``` here? Because by the current logic, if values is null, you will get a NPE at https://github.com/shuyangzhou/liferay-portal/pull/9917/commits/40e9e661f37e91007777f1ce3dee22b051f8829f#diff-39eafb03fec7101df60be56569e519a5d70deac7b42d146d6bc636ee90917554R231

Reply:
Yes, I'll fix and resend. I think I copied that over from replaceKeywords() and didn't fully update it. Thanks!